tools/source-map-retriever-stack.js

```diff
@@ -0,0 +1,60 @@
+import sourceMapSupport from "source-map-support";
+import fs from "fs";
+
+// Why this file exists:
+// We have two places in the tool where we need to do source maps:
+// 1. Loaded isopacks, which use a special custom source map cache
+// 2. Transpiled tool code from Babel
+// 
+// In order to avoid crazy bootstrapping, it would be nice to be able to add
+// functions to look for source maps, so that we can call
+// sourceMapSupport.install as early as possible, and not worry about having
+// the right data structures around.
+// 
+// This module maintains a stack of source map retrieval functions, which are
+// called in reverse order until one returns a truthy value.
+
+var stack = [];
+
+// Add a function to locate source maps; all of the functions are executed in
+// reverse order
+export function push(func) {
+  stack.push(func);
+}
+
+function tryAllSourceMapRetrievers(filename) {
+  for (var i = stack.length - 1; i >= 0; i--) {
+    var sourceMapData = stack[i](filename);
+
+    if (sourceMapData) {
+      return sourceMapData;
+    }
+  }
+
+  return null;
+}
+
+sourceMapSupport.install({
+  retrieveSourceMap: tryAllSourceMapRetrievers,
+  // For now, don't fix the source line in uncaught exceptions, because we
+  // haven't fixed handleUncaughtExceptions in source-map-support to properly
+  // locate the source files.
+  handleUncaughtExceptions: false
+});
+
+// Default retrievers
+
+// Always fall back to the default in the end
+push(sourceMapSupport.retrieveSourceMap);
+
+// Look in <filename>.map
+push((filename) => {
+  var withMapExtension = filename + ".map";
+  if (fs.existsSync(withMapExtension)) {
+    return {
+      map: fs.readFileSync(withMapExtension)
```

Comment:
I think this `map` is supposed to be either a string or a `JSON.parse`d object ([see here](https://github.com/evanw/node-source-map-support/blob/e0e8653cc85008d265051641dd82865d29fa5f83/source-map-support.js#L91-L93)), so this should probably be `fs.readFileSync(withMapExtension, "utf8")` (or it will be a `Buffer`).